include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,480 @@
+    *                      std::ranges::forward_range.
+    * \param[in] w_        The number of values in one window.
+    */
+    template <typename rng_t>
+    //!\cond
+     requires std::ranges::viewable_range<rng_t> &&
+              std::constructible_from<urng_t, ranges::ref_view<std::remove_reference_t<rng_t>>>
+    //!\endcond
+    minimiser_view(rng_t && urange_, uint32_t const w_) :
+    urange{std::views::all(std::forward<rng_t>(urange_))}, window_values_size{w_} {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the range.
+     * \returns Iterator to the first element.
+     *
+     * \details
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee.
+     */
+    auto begin()
+    {
+        return window_iterator<urng_t>{std::ranges::begin(urange), std::ranges::end(urange), window_values_size};
+    }
+
+    //!\copydoc begin()
+    auto begin() const
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return window_iterator<urng_t const>{std::ranges::begin(urange), std::ranges::end(urange), window_values_size};
+    }
+
+    //!\copydoc begin()
+    auto cbegin() const
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * \details
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto end()
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto end() const
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto cend() const
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return end();
+    }
+    //!\}
+};
+
+//!\brief Iterator for calculating minimisers.
+template <std::ranges::view urng_t>
+template <typename rng_t>
+class minimiser_view<urng_t>::window_iterator
+{
+private:
+    //!\brief The iterator type of the underlying range.
+    using it_t = std::ranges::iterator_t<rng_t>;
+    //!\brief The sentinel type of the underlying range.
+    using sentinel_t = std::ranges::sentinel_t<rng_t>;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Type for distances between iterators.
+    using difference_type = typename std::iter_difference_t<it_t>;
+    //!\brief Value type of this iterator.
+    using value_type = reference_t<urng_t>;
```

Comment:
```suggestion
    using value_type = std::ranges::range_value_t<urng_t>;
```
we don't want the value type to have a reference and the use the ranges function not our own (Wiep is changing that)